backend/internal/api/api.go

```diff
@@ -40,10 +37,11 @@ func (h Handler) EndpointInitialize(mux *mux.Router) {
 	mux.HandleFunc(h.getAllEndpoint, h.DBGetAllHandler).Methods("GET")
 	mux.HandleFunc(h.getEndpoint, h.DBGetHandler).Methods("GET")
 	mux.HandleFunc(h.postEndpoint, h.DBPostHandler).Methods("POST")
+
 }
 
 // NewHandler returns handler for database manager.
-func NewHandler(dbManager DBManager, idGenerator IdGenerator) Handler {
+func NewHandler(dbManager DBManager, idGenerator utils.IdGenerator) Handler {
```

Comment:
We should consider defining an interface for sending events and passing it in `NewHandler`. Currently `Handler` incorporated logic for creating eventing component. This not bad thing, but has consequences for testing. I reckon you need to rething the overall design a bit.